lucene/core/src/java/org/apache/lucene/search/IndriAndScorer.java

```diff
@@ -0,0 +1,59 @@
+package org.apache.lucene.search;
+
+import java.io.IOException;
+import java.util.List;
+
+/** Combines scores of subscorers. If a subscorer does not contain
+ *  the docId, a smoothing score is calculated for that 
+ *  document/subscorer combination.
+ */
+public class IndriAndScorer extends IndriDisjunctionScorer {
```

Comment:
Do we expect to have other scorers that need to extend the `IndriDisjunctionScorer`?  Here we have only one such subclass, so I wonder if we could just fold all of this into a single `IndriDisjunctionScorer` class?  Or, are you laying bedrock for future additional query types?

Reply:
Yes, I do hope to be able to add additional Indri query types that will extend the IndriDisjuctionScorer.